### DISCUSSION TITLE
🤔 How to return user data along with the token in `/auth/callback` JWT + Bearer?

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi-users/fastapi-users#1381] [Q&A] [answered]
Hi 👋🏻 

First of all, Thanks for this amazing package. This is super useful and I'm trying to make contribution as I can.

I've question while implementing the OAuth only full stack app that use SSR app ([NuxtJS](https://nuxt.com/)) on frontend and FastAPI User on backend. I've implmented google auth and it's doing great except after successful API call to `/auth/callback`  I've to make another API call for fetching the user data. I dive into the code and found that `/auth/callback` actually has the user data so what will be elegant way to send user data along with bearer token to avoid extra API call?

Thanks 🥂

**Comment by frankie567** [comment, marked as answer]
The thing is, the OAuth callback route is designed to return the response generated by the authentication backend's [`Transport`](https://fastapi-users.github.io/fastapi-users/latest/configuration/authentication/#transport), by calling the `get_login_response` method. Problem is this method only receives the token, not the user.

So, if you *really* want to do this, you would need to reimplement the `callback` endpoint to fit it to your needs: https://github.com/fastapi-users/fastapi-users/blob/61dba2694c3c03dd0fca2a6407bf1f42c6726027/fastapi_users/router/oauth.py#L101
